core/Plugin/Visualization.php

```diff
@@ -600,6 +602,17 @@ private function hasReportBeenPurged()
         return PrivacyManager::hasReportBeenPurged($this->dataTable);
     }
 
+
+    private function hasReportSegmentDisabled()
+    {
+        $module = $this->requestConfig->getApiModuleToRequest();
+        var_dump($module);
+        if (Rules::isSegmentPluginArchivingDisabled($module)) {
```

Comment:
You also need to check if a segment was actually applied to this report, Otherwise it will also be shown for unsegmented reports, that have no data for other reasons.